flink/src/main/java/org/apache/iceberg/flink/source/FlinkSource.java

```diff
@@ -197,7 +206,7 @@ public FlinkInputFormat buildFormat() {
       TypeInformation<RowData> typeInfo = RowDataTypeInfo.of(FlinkSchemaUtil.convert(context.project()));
 
       if (!context.isStreaming()) {
-        return env.createInput(format, typeInfo);
+        return createInputDataStream(format, context, typeInfo);
```

Comment:
In this [comment](https://github.com/apache/iceberg/pull/1936#discussion_r554954082), I think I did not describe the things  clearly.   I mean  we could move the `inferParallelism` into a separate method, don't have to contains the DataStream constructing or chaining methods.

```java
private int inferParallelism(FlinkInputFormat format, ScanContext context) {
   // ....
}
```